### ISSUE TITLE
Support for variable assignment to add variables to the map.

### ISSUE THREAD

**Issue by bamos** [bamos/cpp-expression-parser#3]

**Comment by bamos** [comment]
Also resolved

**Comment by VinGarcia** [comment]
This one is not fully resolved yet @bamos, we can add support to it by adding a function "set_value()" but we still don't have an working operator "=". I have made an experiment to try to add it (saved on a separated branch) but I ended up with one problem and one choice:

problem:

`a = b = c` should be evaluated from right to left, I must think how to make that happen.

choice:

I need to make a new Token called rToken that not only has a value, but also keeps a reference to its origin, or else when trying to assign to an existing variable we would replace its reference for its value before the assignment.

I though of 3 ways of doing this second part:
1. Keep a pointer to the referred packToken, it would be like a javascript closure where the variable changed is always the same independent of the current scope. But it comes with a problem: If the packToken is erased from the map we have a seg. fault in hands =/.
2. Keep only the variable name as a reference, so we can add it to whichever namespace is in the current scope.

I will try to implement first the second approach, since is simpler, but it would be cool to have a variable system like javascript.
